qutebrowser/completion/models/miscmodels.py

```diff
@@ -17,252 +17,132 @@
-
-    COLUMN_WIDTHS = (20, 60, 20)
-
-    def __init__(self, parent=None):
-        super().__init__(parent)
-        cmdlist = _get_cmd_completions(include_aliases=True,
-                                       include_hidden=False)
-        cat = self.new_category("Commands")
-        for (name, desc, misc) in cmdlist:
-            self.new_item(cat, name, desc, misc)
-
-
-class HelpCompletionModel(base.BaseCompletionModel):
 
+def helptopic():
     """A CompletionModel filled with help topics."""
+    model = base.CompletionModel()
+
+    cmdlist = _get_cmd_completions(include_aliases=False, include_hidden=True,
+                                   prefix=':')
+    cat = model.new_category("Commands")
+    for (name, desc, misc) in cmdlist:
+        model.new_item(cat, name, desc, misc)
+
+    cat = model.new_category("Settings")
+    for sectname, sectdata in configdata.DATA.items():
+        for optname in sectdata:
+            try:
+                desc = sectdata.descriptions[optname]
+            except (KeyError, AttributeError):
+                # Some stuff (especially ValueList items) don't have a
+                # description.
+                desc = ""
+            else:
+                desc = desc.splitlines()[0]
+            name = '{}->{}'.format(sectname, optname)
+            model.new_item(cat, name, desc)
+    return model
 
-    # https://github.com/qutebrowser/qutebrowser/issues/545
-    # pylint: disable=abstract-method
-
-    COLUMN_WIDTHS = (20, 60, 20)
-
-    def __init__(self, parent=None):
-        super().__init__(parent)
-        self._init_commands()
-        self._init_settings()
-
-    def _init_commands(self):
-        """Fill completion with :command entries."""
-        cmdlist = _get_cmd_completions(include_aliases=False,
-                                       include_hidden=True, prefix=':')
-        cat = self.new_category("Commands")
-        for (name, desc, misc) in cmdlist:
-            self.new_item(cat, name, desc, misc)
-
-    def _init_settings(self):
-        """Fill completion with section->option entries."""
-        cat = self.new_category("Settings")
-        for sectname, sectdata in configdata.DATA.items():
-            for optname in sectdata:
-                try:
-                    desc = sectdata.descriptions[optname]
-                except (KeyError, AttributeError):
-                    # Some stuff (especially ValueList items) don't have a
-                    # description.
-                    desc = ""
-                else:
-                    desc = desc.splitlines()[0]
-                name = '{}->{}'.format(sectname, optname)
-                self.new_item(cat, name, desc)
-
-
-class QuickmarkCompletionModel(base.BaseCompletionModel):
 
+def quickmark():
     """A CompletionModel filled with all quickmarks."""
+    model = base.CompletionModel()
+    model = sqlmodel.SqlCompletionModel(column_widths=(30, 70, 0))
+    model.new_category('Quickmarks')
```

Comment:
The `quickmark-manager` creates and manipulates a table called `Quickmarks`. This creates a query over the table called `Quickmarks`. Kinda looks like sorcery from this angle, but I think the overall design makes more sense than my old version where the completion models themselves were responsible for inserting data.

I guess another option would be to have `SqlTable` objects generate their own query models, so you would have `model.new_category(objreg.get('quickmark-manager').query()). But that seems worse, and involves more `objreg`-staining. Of course, there's gonna be global state no matter what using SQL.